pkgs/servers/http/bozohttpd/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, stdenv
+, fetchurl
+, bmake
+, groff
+, inetutils
+, wget
+, openssl
+, userSupport ? true
+, cgiSupport ? true
+, dirIndexSupport ? true
+, dynamicContentSupport ? true
+, sslSupport ? true
+, luaSupport ? true
+, lua
+, htpasswdSupport ? true
+}:
+
+let inherit (lib) optional optionals;
+in
+stdenv.mkDerivation rec {
+  pname = "bozohttpd";
+  version = "20210227";
+
+  # bozohttpd is developed in-tree in pkgsrc, canonical hashes can be found at:
+  # http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/www/bozohttpd/distinfo
+  src = fetchurl {
+    url = "http://www.eterna.com.au/${pname}/${pname}-${version}.tar.bz2";
+    sha512 = "b838498626ffb7f7e84f31611e0e99aaa3af64bd9376e1a13ec16313c182eebfd9ea2c2d03904497239af723bf34a3d2202dac1f2d3e55f9fd076f6d45ccfa33";
+  };
+
+  # backport two unreleased commits to fix builds on non-netbsd platforms.
+  patches = [
+    # add missing `#include <stdint.h>`
+    # https://freshbsd.org/netbsd/src/commit/qMGNoXfgeieZBVRC
+    ./0001-include-stdint.h.patch
+
+    # BUFSIZ is not guaranteed to be large enough
+    # https://freshbsd.org/netbsd/src/commit/A4ueIHIp3JgjNVRC
+    ./0002-dont-use-host-BUFSIZ.patch
+  ];
+  patchFlags = [ "-p3" ];
+
+  buildInputs = [ openssl ] ++ optional (luaSupport) lua;
+  nativeBuildInputs = [ bmake groff ];
+
+  COPTS =
+    [
+      "-DNO_BLOCKLIST_SUPPORT" # unpackaged dependency: https://man.netbsd.org/blocklist.3
+    ]
+    ++ optional (!userSupport) "-DNO_USER_SUPPORT"
+    ++ optional (!dirIndexSupport) "-DNO_DIRINDEX_SUPPORT"
+    ++ optional (!dynamicContentSupport) "-DNO_DYNAMIC_CONTENT"
+    ++ optional (!luaSupport) "-DNO_LUA_SUPPORT"
+    ++ optional (!sslSupport) "-DNO_SSL_SUPPORT"
+    ++ optional (!cgiSupport) "-DNO_CGIBIN_SUPPORT"
+    ++ optional (htpasswdSupport) "-DDO_HTPASSWD";
```

Comment:
```suggestion
  COPTS = [
    "-DNO_BLOCKLIST_SUPPORT" # unpackaged dependency: https://man.netbsd.org/blocklist.3
    ] ++ optional (!userSupport) "-DNO_USER_SUPPORT"
    ++ optional (!dirIndexSupport) "-DNO_DIRINDEX_SUPPORT"
    ++ optional (!dynamicContentSupport) "-DNO_DYNAMIC_CONTENT"
    ++ optional (!luaSupport) "-DNO_LUA_SUPPORT"
    ++ optional (!sslSupport) "-DNO_SSL_SUPPORT"
    ++ optional (!cgiSupport) "-DNO_CGIBIN_SUPPORT"
    ++ optional htpasswdSupport "-DDO_HTPASSWD";
```

Reply:
While I don't entirely disagree with this, `nixpkgs-fmt` does. Do you use a different formatting tool?

Reply:
nixpkgs-fmt does not enforce styling. It will happily accept both variants.

Reply:
Right, `nixpkgs-fmt` will accept both what I've done here:

```
VAR =
  [
    a
  ]
  ++ b
  ++ c;
```

and the annoyingly asymmetric:

```
VAR = [
  a
] ++ b
++ c;
```

but not what you suggest:

```
VAR = [
    a
  ] ++ b
  ++ c;
```

or what I'd personally prefer:


```
VAR = [
    a
  ]
  ++ b
  ++ c;
```

...hence my question about if you were using a different formatting tool which does accept this suggestion, or if you just typed it into the web UI by hand without checking.